docs/docs/troubleshooting/hardware-issues.mdx

```diff
@@ -0,0 +1,183 @@
+
+For through-hole soldering specifically, the below infographic can be quite helpful.
+![A helpful soldering infographic. Source: Sparkfun](https://cdn.sparkfun.com/assets/c/d/a/a/9/523b1189757b7fb36e8b456b.jpg)
+:::tip
+The above image was taken from [Sparkfun's guide on soldering through-hole components](https://learn.sparkfun.com/tutorials/how-to-solder-through-hole-soldering), which may be of use if you are less experienced with soldering.
+:::
+
+</TabItem>
+</PinTabs>
+
+### Resolving Issues
+
+Many issues can be resolved by reworking the affected solder joints (Look up a tutorial on how to do so for your component). Sometimes a pin ends up broken though, in which case you will need to "bodge" (solder) a wire from a spare GPIO pin to the connection in need of a working pin.
+
+Once you have done so, you will need to adjust the `kscan` of your keyboard slightly. The recommended approach to doing so is via your `zmk-config`:
+
+1. Search through the files defining your board/shield for the definition of the keyboard's `kscan`. This is typically a`.dts`, `.dtsi`, or an `.overlay` file.
+2. Make a file called `<my_keyboard>.overlay` file in your `zmk-config`, where `my_keyboard` is the name of the affected keyboard (part).
+3. Add a [phandle](https://docs.zephyrproject.org/latest/build/dts/phandles.html) to this new file pointing at your `kscan`, copying in the property of your `kscan` where the broken GPIO pin appears. Depending on your `kscan` driver this could be `input-gpios`, `row-gpios`, etc.
+4. Edit the node to replace the old GPIO key with the new. For example:
+
+```dts
+&kscan0 {
+        input-gpios
+        = <&pro_micro  7 (GPIO_ACTIVE_LOW | GPIO_PULL_UP)>
+        , <&pro_micro  6 (GPIO_ACTIVE_LOW | GPIO_PULL_UP)>
+        , <&pro_micro  9 (GPIO_ACTIVE_LOW | GPIO_PULL_UP)>
+        ;
+    };
+```
+
+could have the pin `pro_micro 6` replaced with `gpio0 8`.
+
+```dts title="<my_keyboard>.overlay"
+&kscan0 {
+        input-gpios
+        = <&pro_micro  7 (GPIO_ACTIVE_LOW | GPIO_PULL_UP)>
+        , <&gpio0  8 (GPIO_ACTIVE_LOW | GPIO_PULL_UP)>
+        , <&pro_micro  9 (GPIO_ACTIVE_LOW | GPIO_PULL_UP)>
+        ;
+    };
+```
+
+You can then build and flash using your config as usual.
+
+## Antenna
+
+Under ideal conditions, a Bluetooth connection can be stable at even a 100 meter distance. However, there are many things which can affect the effective range of the antenna. In no particular order, the most common issues you will encounter are:
+
+- Solid material near an antenna or between the two communicating antennas (Metal is _particularly_ bad)
+- Interference from other nearby 2.4GHz connections
+- The antenna makes use of malfunctioning hardware such as a [faulty oscillator](connection-issues.mdx#mitigating-a-faulty-oscillator)
+- Not enough [power being provided to the antenna](connection-issues.mdx#unreliableweak-connection)
+
```

Comment:
We might want to note one of the most common cause of such issues here: missing external antennas for a motherboard or a PCIe wireless chipset.

Another more obscure cause (related to point 2): https://www.bluetoothandusb3.com/the-explanation